src/main/java/services/impl/FormRecordProcessorImpl.java

```diff
@@ -112,6 +139,13 @@ public static void purgeCases(UserSqlSandbox sandbox) {
         log.info(String.format(
                 "Purged [%d Case, %d Ledger] records in %dms",
                 removedCaseCount, removedLedgers, taken));
-
+        datadogStatsDClient.recordExecutionTime(
+                Constants.DATADOG_TIMINGS,
+                taken,
+                "domain:" + restoreFactory.getDomain(),
+                "username" + restoreFactory.getWrappedUsername(),
+                "request:" + "case-purge",
+                "case-count" + removedCaseCount
```

Comment:
The format of tags must be 'name:value' (username and case-count are missing ':')

It's not useful to have random numerical values as tags. If you want to have a case-count tag you should bucket.

Is it necessary to have `username` tag - just that it will push up the metric count a lot: https://help.datadoghq.com/hc/en-us/articles/204271775-What-is-a-custom-metric-and-what-is-the-limit-on-the-number-of-custom-metrics-I-can-have-

Reply:
I'd also add that `username` is not likely to be a very useful tag in graphs if there are a high volume of users on your graph. I've worked around this in the past by logging that information to a file rather than sending to datadog.

Another related thing I've done is only tag certain values (like if you only care about certain usernames, put them in a whitelist and only add that tag if the username is in the list). We do that for `domain` in HQ since tagging every domain would push up the custom metric count a lot, but there are certain key domains that we'd like to have tagged (`rec`, for example).

Reply:
Ah, missed @dannyroberts comment above. He said it better.